From 416b8c1f0db109b9e7849d203c1cb5c14924008c Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Tue, 20 Jan 2015 23:55:50 -0500
Subject: [PATCH] add Settings.Global key used by Performance->DisableDimmer

---
 core/java/android/provider/Settings.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 3da43fd..cb51aa9 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -5299,6 +5299,12 @@ public final class Settings {
         public static final String MAX_GPU_SPEED = "max_gpu_speed";
 
         /**
+         * Whether to disable Nvidia 'SmartDimmer'
+         * @hide
+         */
+        public static final String DISABLE_SMARTDIMMER = "disable_smartdimmer";
+
+        /**
          * Whether ADB is enabled.
          */
         public static final String ADB_ENABLED = "adb_enabled";
-- 
1.9.1

