From 0a81c7182cc4b66bb388b54dd712f47c17a8c5c2 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Wed, 3 Dec 2014 12:38:12 -0500
Subject: [PATCH] fix FC when opening settings pages

---
 .../android/browser/BrowserPreferencesPage.java    |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/browser/BrowserPreferencesPage.java b/src/com/android/browser/BrowserPreferencesPage.java
index 6e6da17..766b653 100644
--- a/src/com/android/browser/BrowserPreferencesPage.java
+++ b/src/com/android/browser/BrowserPreferencesPage.java
@@ -98,4 +98,9 @@ public class BrowserPreferencesPage extends PreferenceActivity {
         return intent;
     }
 
+    @Override
+    protected boolean isValidFragment(String fragmentName) {
+        return true;
+    }
+
 }
-- 
1.7.9.5

