From 45116f745e07c8b99a954693edff86f1a5f8ddb3 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Wed, 12 Mar 2014 23:13:07 -0500
Subject: [PATCH] prevent full bootup when attaching AC adapter

---
 init/init.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/init/init.c b/init/init.c
index bd1db7a..05f9e9a 100644
--- a/init/init.c
+++ b/init/init.c
@@ -736,6 +736,15 @@ static void import_kernel_nv(char *name, int for_emulator)
         cnt = snprintf(prop, sizeof(prop), "ro.boot.%s", boot_prop_name);
         if (cnt < PROP_NAME_MAX)
             property_set(prop, value);
+
+    /* DoPa (20140312) - the Hisense bootloader fails to use 'androidboot.mode=charger'
+     * to indicate that the tablet is booting solely to show the battery animations;
+     * this workaround looks for a particular value of 'power_reason' that corresponds
+     * to that state (Hisense appears to have done something similar in their version
+     * of 'init').
+     */
+    } else if (!strcmp(name, "power_reason") && !strcmp(value, "0x808800")) {
+        property_set("ro.boot.mode", "charger");
     }
 }
 
-- 
1.7.9.5

