From 1f6ba1f352a6470bdc8ed209245c5d4846194247 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Fri, 27 Jun 2014 02:04:49 -0400
Subject: [PATCH] remove 'ril-daemon' and 'flash_recovery' services from
 init.rc

---
 rootdir/init.rc |   12 ------------
 1 file changed, 12 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index cbcb842..c913c9d 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -540,13 +540,6 @@ service debuggerd /system/bin/debuggerd
 service debuggerd64 /system/bin/debuggerd64
     class main
 
-service ril-daemon /system/bin/rild
-    class main
-    socket rild stream 660 root radio
-    socket rild-debug stream 660 radio system
-    user root
-    group radio cache inet misc audio log
-
 service surfaceflinger /system/bin/surfaceflinger
     class core
     user system
@@ -589,11 +582,6 @@ service installd /system/bin/installd
     class main
     socket installd stream 600 system system
 
-service flash_recovery /system/bin/install-recovery.sh
-    class main
-    seclabel u:r:install_recovery:s0
-    oneshot
-
 service racoon /system/bin/racoon
     class main
     socket racoon stream 600 system system
-- 
1.7.9.5

