From a1dd5d8f0d22f38e356e6dabf5048755dcf1c8b4 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Wed, 12 Mar 2014 23:00:00 -0500
Subject: [PATCH] add more detail to dmesg log for init operations

---
 rootdir/init.rc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index c913c9d..149bbd3 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -32,7 +32,7 @@ on early-init
 on init
     sysclktz 0
 
-    loglevel 3
+    loglevel 6
 
     # Backward compatibility
     symlink /system/etc /etc
-- 
1.7.9.5

