From 4c17c8218568d96ce4c070ab0dd83b487c052204 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Sun, 24 Aug 2014 19:19:39 -0400
Subject: [PATCH] reduce AAPT spam

---
 core/package_internal.mk |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/package_internal.mk b/core/package_internal.mk
index a6ad354..e0f89f9 100644
--- a/core/package_internal.mk
+++ b/core/package_internal.mk
@@ -63,10 +63,10 @@ ifeq ($(LOCAL_MODULE_TAGS),)
 LOCAL_MODULE_TAGS := optional
 endif
 
-ifeq ($(filter tests, $(LOCAL_MODULE_TAGS)),)
+#ifeq ($(filter tests, $(LOCAL_MODULE_TAGS)),)
 # Force localization check if it's not tagged as tests.
-LOCAL_AAPT_FLAGS := $(LOCAL_AAPT_FLAGS) -z
-endif
+#LOCAL_AAPT_FLAGS := $(LOCAL_AAPT_FLAGS) -z
+#endif
 
 ifdef LOCAL_PACKAGE_SPLITS
 LOCAL_AAPT_FLAGS += $(addprefix --split ,$(LOCAL_PACKAGE_SPLITS))
-- 
1.7.9.5

