From 4e9c4cc86224ffff2831012e2c0b00ead309d740 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Sat, 15 Nov 2014 00:12:25 -0500
Subject: [PATCH] lollipop: change OTA package from blocks to files

---
 core/Makefile |    1 -
 1 file changed, 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index 31f51eb..43d10c1 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1482,7 +1482,6 @@ $(INTERNAL_OTA_PACKAGE_TARGET): $(BUILT_TARGET_FILES_PACKAGE) $(DISTTOOLS)
 	@echo "Package OTA: $@"
 	$(hide) MKBOOTIMG=$(MKBOOTIMG) \
 	   ./build/tools/releasetools/ota_from_target_files -v \
-	   --block \
 	   -p $(HOST_OUT) \
 	   -k $(KEY_CERT_PAIR) \
 	   $(OTA_EXTRA_OPTIONS) \
-- 
1.7.9.5

