From 744ffc59bd0fb0b57e10505b85df55389cec2b74 Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Thu, 18 Dec 2014 12:36:50 -0500
Subject: [PATCH] add 'ifdef __cplusplus' to f2fs_fs.h

---
 include/f2fs_fs.h |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/f2fs_fs.h b/include/f2fs_fs.h
index b02002c..58b9ec2 100644
--- a/include/f2fs_fs.h
+++ b/include/f2fs_fs.h
@@ -20,6 +20,10 @@
 #include <config.h>
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef u_int64_t	u64;
 typedef u_int32_t	u32;
 typedef u_int16_t	u16;
@@ -777,4 +781,7 @@ f2fs_hash_t f2fs_dentry_hash(const unsigned char *, int);
 
 extern struct f2fs_configuration config;
 
+#ifdef __cplusplus
+}
+#endif
 #endif	/*__F2FS_FS_H */
-- 
1.7.9.5

