From bb61584ce1847498b5805b2d58333c333a158eca Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Sat, 15 Nov 2014 00:05:38 -0500
Subject: [PATCH] replace unsupported wait3() with waitpid()

---
 miscutils/time.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/miscutils/time.c b/miscutils/time.c
index 19b0b44..9e47e07 100644
--- a/miscutils/time.c
+++ b/miscutils/time.c
@@ -71,7 +71,10 @@ static void resuse_end(pid_t pid, resource_t *resp)
 
 	/* Ignore signals, but don't ignore the children.  When wait3
 	 * returns the child process, set the time the command finished. */
-	while ((caught = wait3(&resp->waitstatus, 0, &resp->ru)) != pid) {
+
+/*	while ((caught = wait3(&resp->waitstatus, 0, &resp->ru)) != pid) { */
+
+	while ((caught = waitpid(-1, &resp->waitstatus, 0)) != pid) {
 		if (caught == -1 && errno != EINTR) {
 			bb_perror_msg("wait");
 			return;
-- 
1.7.9.5

