From 3a4e15e70cfb4bea02a94608b271cff861a4fa2c Mon Sep 17 00:00:00 2001
From: DoloresPark <dolorespark@e-vertise.com>
Date: Tue, 26 Aug 2014 11:14:00 -0400
Subject: [PATCH] disable if building AOSP recovery

---
 Android.mk |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Android.mk b/Android.mk
index 8c9d3d6..59097c8 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,5 +1,7 @@
 LOCAL_PATH:= $(call my-dir)
 
+ifneq ($(RECOVERY_VARIANT),)
+
 ifneq ($(BOARD_VOLD_MAX_PARTITIONS),)
 common_cflags += -DVOLD_MAX_PARTITIONS=$(BOARD_VOLD_MAX_PARTITIONS)
 endif
@@ -131,3 +133,5 @@ LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
 LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_EXECUTABLE)
+
+endif
-- 
1.7.9.5

